dracut/image-boot/eos-image-boot-generator

```diff
@@ -0,0 +1,86 @@
+#!/bin/sh
+# Copyright (C) 2016 Endless Mobile, Inc.
+# Licensed under the GPLv2
+#
+# Support booting from an image file hosted on a filesystem.
+# We parse the kernel parameters requesting this functionality and configure
+# other systemd units to go about setting up the block devices and mounting
+# the target filesystem.
+
+type getarg >/dev/null 2>&1 || . /lib/dracut-lib.sh
+
+GENERATOR_DIR="$1"
+
+image_device=$(getarg endless.image.device=)
+image_path=$(getarg endless.image.path=)
+
+[ -z "${image_device}" -o -z "${image_path}" ] && exit 0
+
+case "${image_device}" in
+PARTUUID=*)
+  image_device=/dev/disk/by-partuuid/${image_device#PARTUUID=}
```

Comment:
Might it be simpler to pass `endless.image.device=/dev/disk/by-partuuid/5fda738f-765f-4c38-a3ab-bba8b62e2b90` directly from Grub, rather than inventing and handling `PARTUUID=`, `UUID=`? I appreciate that passing the full path from Grub would work with this branch as it stands; and I'm happy to be disagreed with on this. Just a thought.